lib/home_page.dart

```diff
@@ -0,0 +1,348 @@
+  final _opinionController = TextEditingController();
+  bool _acceptTerms = false;
+  Gender? _selectedGender;
+  Country? _selectedCountry;
+
+  void _submitForm() {
+    if (_formKey.currentState!.validate()) {
+      _formKey.currentState?.save();
+      showDialog(
+        context: context,
+        builder: (BuildContext context) => _createAlertDialog(),
+      );
+    }
+  }
+
+  @override
+  void dispose() {
+    nameController.dispose();
+    _opinionController.dispose();
+    _dobController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    var headlineSmall = Theme.of(context).textTheme.headlineSmall;
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(
+          appBarTitle,
+          style: headlineSmall,
+        ),
+      ),
+      body: Padding(
+        padding: screenPadding,
+        child: Form(
+          key: _formKey,
+          child: SingleChildScrollView(
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                _createTextFormFieldForName(),
+                sizedBoxFormPage,
+                _createTextFormFieldForDob(),
+                sizedBoxFormPage,
+                _createRadioListTileForGender(),
+                sizedBoxFormPage,
+                _createTextFormFieldForOpinion(),
+                sizedBoxFormPage,
+                _createDropDownFormFieldForCountry(),
+                sizedBoxFormPage,
+                _createCheckBoxForTermsAndConditions(),
+                sizedBoxFormPage,
+                _createSubmitButton(),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  RadioListTile _createRadioListTile(Gender gender) {
+    var bodySmall = Theme.of(context).textTheme.bodySmall;
+    return RadioListTile(
+      activeColor: selectedWidgetColor,
+      title: Text(
+        gender.name!,
+        style: bodySmall,
+      ),
+      value: gender,
+      groupValue: _selectedGender,
+      onChanged: (value) {
+        setState(() {
+          _selectedGender = value;
+        });
+      },
+    );
+  }
+
+  DropdownButtonFormField<String> _createDropDownMenu() {
+    var bodySmall = Theme.of(context).textTheme.bodySmall;
+    return DropdownButtonFormField<String>(
+      icon: const Icon(Icons.keyboard_arrow_down),
+      hint: const Text('Select a country'),
+      value: _selectedCountry?.name,
+      onChanged: (String? value) {
+        _selectedCountry = getCountry(value);
+        setState(() {});
+      },
+      items: Country.values.map((Country country) {
+        return DropdownMenuItem<String>(
+          value: country.name,
+          child: Text(
+            country.name ?? '',
+            style: bodySmall,
+          ),
+        );
+      }).toList(),
+    );
+  }
+
+  AlertDialog _createAlertDialog() {
+    var bodySmall = Theme.of(context).textTheme.bodySmall;
```

Comment:
Why var. Please check if this style is as per design.

Reply:
changed to final.